tests/generate-transaction-tests/index.js

```diff
@@ -247,61 +247,70 @@ const createEnvelopeTestcase = (valid) => {
   });
 };
 
-const sampleArguments = (arguments) => {
-  return arguments.map(({ type, sampleValue }) => {
-    return {
-      type: type,
-      value: sampleValue,
-    };
+const sampleArguments = (i, arguments) => {
+  return arguments.map(({ type, sampleValues }) => {
+    return sampleValues[Math.min(i, Object.keys(sampleValues).length-1)];
   });
 };
 
 const testnetTemplates = JSON.parse(fs.readFileSync('manifest.testnet.json')).templates;
 const mainnetTemplates = JSON.parse(fs.readFileSync('manifest.mainnet.json')).templates;
 
-const manifestTestnetPayloadCases = testnetTemplates.map((template) => {
-  return [
-    `${template.id} - ${template.name}`,
+const numberOfRequiredTests = (arguments) => {
+  return Math.max(1, ...arguments.map(({ type, sampleValues }) => Object.keys(sampleValues).length));
```

Comment:
wait - sampleValues is an object or a array? And I am at loss about how this works. It needs some comment. Naturally one would think that this would try to use all combinations so we should document how exactly are tests generated in case you have one argument with 2 values and second argument with three values.

Reply:
I don't want to do all combinations. First, as of now there is always at most one argument with multiple samplevalues, Second, Doing all combinations would create too many test cases. For zemu tests you need to visually check the generated output. I believe it would be counterproductive to have too many tests like this. Besides that, the test are already slow enough. I added a comment to explain what we generate.

Reply:
sampleValues is an array.